File: src/main/java/org/codelibs/fess/ds/salesforce/api/sobject/SDandBCompany.java
/*
 * Copyright 2012-2019 CodeLibs Project and the Others.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */
package org.codelibs.fess.ds.salesforce.api.sobject;

import java.util.Arrays;
import java.util.List;

import org.codelibs.fess.ds.salesforce.api.SearchLayout;

/** D&amp;B 企業 */
public class SDandBCompany extends SearchLayout {

    /** D&amp;B 企業 ID */
    protected static final String ID = "Id";
    /** 第 1 会社名 */
    protected static final String NAME = "Name";
    /** 作成日 */
    protected static final String CREATED_DATE = "CreatedDate";
    /** 最終更新日 */
    protected static final String LAST_MODIFIED_DATE = "LastModifiedDate";
    /** System Modstamp */
    protected static final String SYSTEM_MODSTAMP = "SystemModstamp";
    /** 最終閲覧日 */
    protected static final String LAST_VIEWED_DATE = "LastViewedDate";
    /** 最終参照日 */
    protected static final String LAST_REFERENCED_DATE = "LastReferencedDate";
    /** D-U-N-S number */
    protected static final String DUNS_NUMBER = "DunsNumber";
    /** 住所 */
    protected static final String STREET = "Street";
    /** 市区郡 */
    protected static final String CITY = "City";
    /** 都道府県 */
    protected static final String STATE = "State";
    /** 郵便番号 */
    protected static final String POSTAL_CODE = "PostalCode";
    /** 国 */
    protected static final String COUNTRY = "Country";
    /** Geocode Accuracy */
    protected static final String GEOCODE_ACCURACY_STANDARD = "GeocodeAccuracyStandard";
    /** 電話番号 */
    protected static final String PHONE = "Phone";
    /** Fax 番号 */
    protected static final String FAX = "Fax";
    /** 国際ダイヤルコード */
    protected static final String COUNTRY_ACCESS_CODE = "CountryAccessCode";
    /** 所有者種別インジケータ */
    protected static final String PUBLIC_INDICATOR = "PublicIndicator";
    /** 株式コード */
    protected static final String STOCK_SYMBOL = "StockSymbol";
    /** 証券取引所 */
    protected static final String STOCK_EXCHANGE = "StockExchange";
    /** URL */
    protected static final String URL = "URL";
    /** 倒産インジケータ */
    protected static final String OUT_OF_BUSINESS = "OutOfBusiness";
    /** FIPS MSA コード */
    protected static final String FIPS_MSA_CODE = "FipsMsaCode";
    /** FIPS MSA コードの説明 */
    protected static final String FIPS_MSA_DESC = "FipsMsaDesc";
    /** 第 1 取引形態 */
    protected static final String TRADE_STYLE1 = "TradeStyle1";
    /** 設立年 */
    protected static final String YEAR_STARTED = "YearStarted";
    /** 町名・番地(郵送先) */
    protected static final String MAILING_STREET = "MailingStreet";
    /** 市区郡(郵送先) */
    protected static final String MAILING_CITY = "MailingCity";
    /** 都道府県(郵送先) */
    protected static final String MAILING_STATE = "MailingState";
    /** 取引先責任者 郵便番号(郵送先) */
    protected static final String MAILING_POSTAL_CODE = "MailingPostalCode";
    /** 国(郵送先) */
    protected static final String MAILING_COUNTRY = "MailingCountry";
    /** Mailing Geocode Accuracy */
    protected static final String MAILING_GEOCODE_ACCURACY = "MailingGeocodeAccuracy";
    /** 緯度 */
    protected static final String LATITUDE = "Latitude";
    /** 経度 */
    protected static final String LONGITUDE = "Longitude";
    /** 第 1 SIC コード */
    protected static final String PRIMARY_SIC = "PrimarySic";
    /** 第 1 SIC の説明 */
    protected static final String PRIMARY_SIC_DESC = "PrimarySicDesc";
    /** 第 2 SIC コード */
    protected static final String SECOND_SIC = "SecondSic";
    /** 第 2 SIC の説明 */
    protected static final String SECOND_SIC_DESC = "SecondSicDesc";
    /** 第 3 SIC コード */
    protected static final String THIRD_SIC = "ThirdSic";
    /** 第 3 SIC の説明 */
    protected static final String THIRD_SIC_DESC = "ThirdSicDesc";
    /** 第 4 SIC コード */
    protected static final String FOURTH_SIC = "FourthSic";
    /** 第 4 SIC の説明 */
    protected static final String FOURTH_SIC_DESC = "FourthSicDesc";
    /** 第 5 SIC コード */
    protected static final String FIFTH_SIC = "FifthSic";
    /** 第 5 SIC の説明 */
    protected static final String FIFTH_SIC_DESC = "FifthSicDesc";
    /** 第 6 SIC コード */
    protected static final String SIXTH_SIC = "SixthSic";
    /** 第 6 SIC の説明 */
    protected static final String SIXTH_SIC_DESC = "SixthSicDesc";
    /** 第 1 NAICS コード */
    protected static final String PRIMARY_NAICS = "PrimaryNaics";
    /** 第 1 NAICS の説明 */
    protected static final String PRIMARY_NAICS_DESC = "PrimaryNaicsDesc";
    /** 第 2 NAICS コード */
    protected static final String SECOND_NAICS = "SecondNaics";
    /** 第 2 NAICS の説明 */
    protected static final String SECOND_NAICS_DESC = "SecondNaicsDesc";
    /** 第 3 NAICS コード */
    protected static final String THIRD_NAICS = "ThirdNaics";
    /** 第 3 NAICS の説明 */
    protected static final String THIRD_NAICS_DESC = "ThirdNaicsDesc";
    /** 第 4 NAICS コード */
    protected static final String FOURTH_NAICS = "FourthNaics";
    /** 第 4 NAICS の説明 */
    protected static final String FOURTH_NAICS_DESC = "FourthNaicsDesc";
    /** 第 5 NAICS コード */
    protected static final String FIFTH_NAICS = "FifthNaics";
    /** 第 5 NAICS の説明 */
    protected static final String FIFTH_NAICS_DESC = "FifthNaicsDesc";
    /** 第 6 NAICS コード */
    protected static final String SIXTH_NAICS = "SixthNaics";
    /** 第 6 NAICS の説明 */
    protected static final String SIXTH_NAICS_DESC = "SixthNaicsDesc";
    /** 場所所有者インジケータ */
    protected static final String OWN_OR_RENT = "OwnOrRent";
    /** 従業員数 - 場所インジケータ */
    protected static final String EMPLOYEES_HERE_RELIABILITY = "EmployeesHereReliability";
    /** 年間売上高インジケータ */
    protected static final String SALES_VOLUME_RELIABILITY = "SalesVolumeReliability";
    /** 現地通貨コード */
    protected static final String CURRENCY_CODE = "CurrencyCode";
    /** 法律上の会社形態 */
    protected static final String LEGAL_STATUS = "LegalStatus";
    /** 従業員数 - 合計インジケータ */
    protected static final String EMPLOYEES_TOTAL_RELIABILITY = "EmployeesTotalReliability";
    /** マイノリティ所有インジケータ */
    protected static final String MINORITY_OWNED = "MinorityOwned";
    /** 女性所有インジケータ */
    protected static final String WOMEN_OWNED = "WomenOwned";
    /** 中小企業インジケータ */
    protected static final String SMALL_BUSINESS = "SmallBusiness";
    /** マーケティングセグメンテーションクラスタ */
    protected static final String MARKETING_SEGMENTATION_CLUSTER = "MarketingSegmentationCluster";
    /** 輸入/輸出 */
    protected static final String IMPORT_EXPORT_AGENT = "ImportExportAgent";
    /** 関連子会社インジケータ */
    protected static final String SUBSIDIARY = "Subsidiary";
    /** 第 2 取引形態 */
    protected static final String TRADE_STYLE2 = "TradeStyle2";
    /** 第 3 取引形態 */
    protected static final String TRADE_STYLE3 = "TradeStyle3";
    /** 第 4 取引形態 */
    protected static final String TRADE_STYLE4 = "TradeStyle4";
    /** 第 5 取引形態 */
    protected static final String TRADE_STYLE5 = "TradeStyle5";
    /** 国民 ID 番号 */
    protected static final String NATIONAL_ID = "NationalId";
    /** 国民 ID 体系 */
    protected static final String NATIONAL_ID_TYPE = "NationalIdType";
    /** 米国納税 ID 番号 */
    protected static final String US_TAX_ID = "UsTaxId";
    /** 地理コードの精度 */
    protected static final String GEO_CODE_ACCURACY = "GeoCodeAccuracy";
    /** 支払遅延リスク */
    protected static final String MARKETING_PRE_SCREEN = "MarketingPreScreen";
    /** グローバル代表企業の D-U-N-S Number */
    protected static final String GLOBAL_ULTIMATE_DUNS_NUMBER = "GlobalUltimateDunsNumber";
    /** グローバル代表企業の会社名 */
    protected static final String GLOBAL_ULTIMATE_BUSINESS_NAME = "GlobalUltimateBusinessName";
    /** 親会社の D-U-N-S Number */
    protected static final String PARENT_OR_HQ_DUNS_NUMBER = "ParentOrHqDunsNumber";
    /** 親会社の会社名 */
    protected static final String PARENT_OR_HQ_BUSINESS_NAME = "ParentOrHqBusinessName";
    /** 国内代表企業の D-U-N-S Number */
    protected static final String DOMESTIC_ULTIMATE_DUNS_NUMBER = "DomesticUltimateDunsNumber";
    /** 国内代表企業の会社名 */
    protected static final String DOMESTIC_ULTIMATE_BUSINESS_NAME = "DomesticUltimateBusinessName";
    /** 場所種別 */
    protected static final String LOCATION_STATUS = "LocationStatus";
    /** 現地通貨の ISO コード */
    protected static final String COMPANY_CURRENCY_ISO_CODE = "CompanyCurrencyIsoCode";
    /** 会社の説明 */
    protected static final String DESCRIPTION = "Description";
    /** S&amp;P 500 */
    protected static final String INCLUDED_IN_SN_P500 = "IncludedInSnP500";
    /** 立地面積の精度 */
    protected static final String PREMISES_MEASURE_RELIABILITY = "PremisesMeasureReliability";
    /** 立地面積単位 */
    protected static final String PREMISES_MEASURE_UNIT = "PremisesMeasureUnit";
    /** 第 1 SIC8 コード */
    protected static final String PRIMARY_SIC8 = "PrimarySic8";
    /** 第 1 SIC8 の説明 */
    protected static final String PRIMARY_SIC8_DESC = "PrimarySic8Desc";
    /** 第 2 SIC8 コード */
    protected static final String SECOND_SIC8 = "SecondSic8";
    /** 第 2 SIC8 の説明  */
    protected static final String SECOND_SIC8_DESC = "SecondSic8Desc";
    /** 第 3 SIC8 コード */
    protected static final String THIRD_SIC8 = "ThirdSic8";
    /** 第 3 SIC8 の説明 */
    protected static final String THIRD_SIC8_DESC = "ThirdSic8Desc";
    /** 第 4 SIC8 コード */
    protected static final String FOURTH_SIC8 = "FourthSic8";
    /** 第 4 SIC8 の説明 */
    protected static final String FOURTH_SIC8_DESC = "FourthSic8Desc";
    /** 第 5 SIC8 コード */
    protected static final String FIFTH_SIC8 = "FifthSic8";
    /** 第 5 SIC8 の説明 */
    protected static final String FIFTH_SIC8_DESC = "FifthSic8Desc";
    /** 第 6 SIC8 コード */
    protected static final String SIXTH_SIC8 = "SixthSic8";
    /** 第 6 SIC8 の説明 */
    protected static final String SIXTH_SIC8_DESC = "SixthSic8Desc";

    protected static final List<String> contents = Arrays.asList(NAME, DUNS_NUMBER, PHONE, FAX, URL, LOCATION_STATUS, DESCRIPTION);

    @Override
    public String getTitle() {
        return NAME;
    }

    @Override
    public List<String> getContents() {
        return contents;
    }

}

File: src/main/kotlin/org/codelibs/fess/ds/salesforce/api/sobject/SDandBCompany.kt
/*
 * Copyright 2012-2019 CodeLibs Project and the Others.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */
package org.codelibs.fess.ds.salesforce.api.sobject

import java.util.*

/** D&B 企業 */
data class SDandBCompany(
        /** D&B 企業 ID */
        override val id: String,
        /** 第 1 会社名 */
        val name: String,
        /** 作成日 */
        override val createdDate: Date,
        /** 最終更新日 */
        override val lastModifiedDate: Date,
        /** System Modstamp */
        val systemModstamp: Date,
        /** 最終閲覧日 */
        val lastViewedDate: Date?,
        /** 最終参照日 */
        val lastReferencedDate: Date?,
        /** D-U-N-S number */
        val dunsNumber: String,
        /** 住所 */
        val street: String?,
        /** 市区郡 */
        val city: String?,
        /** 都道府県 */
        val state: String?,
        /** 郵便番号 */
        val postalCode: String?,
        /** 国 */
        val country: String?,
        /** Geocode Accuracy */
        val geocodeAccuracyStandard: String?,
        /** 電話番号 */
        val phone: String?,
        /** Fax 番号 */
        val fax: String?,
        /** 国際ダイヤルコード */
        val countryAccessCode: String?,
        /** 所有者種別インジケータ */
        val publicIndicator: String?,
        /** 株式コード */
        val stockSymbol: String?,
        /** 証券取引所 */
        val stockExchange: String?,
        /** URL */
        val uRL: String?,
        /** 倒産インジケータ */
        val outOfBusiness: String?,
        /** FIPS MSA コード */
        val fipsMsaCode: String?,
        /** FIPS MSA コードの説明 */
        val fipsMsaDesc: String?,
        /** 第 1 取引形態 */
        val tradeStyle1: String?,
        /** 設立年 */
        val yearStarted: String?,
        /** 町名・番地(郵送先) */
        val mailingStreet: String?,
        /** 市区郡(郵送先) */
        val mailingCity: String?,
        /** 都道府県(郵送先) */
        val mailingState: String?,
        /** 取引先責任者 郵便番号(郵送先) */
        val mailingPostalCode: String?,
        /** 国(郵送先) */
        val mailingCountry: String?,
        /** Mailing Geocode Accuracy */
        val mailingGeocodeAccuracy: String?,
        /** 緯度 */
        val latitude: String?,
        /** 経度 */
        val longitude: String?,
        /** 第 1 SIC コード */
        val primarySic: String?,
        /** 第 1 SIC の説明 */
        val primarySicDesc: String?,
        /** 第 2 SIC コード */
        val secondSic: String?,
        /** 第 2 SIC の説明 */
        val secondSicDesc: String?,
        /** 第 3 SIC コード */
        val thirdSic: String?,
        /** 第 3 SIC の説明 */
        val thirdSicDesc: String?,
        /** 第 4 SIC コード */
        val fourthSic: String?,
        /** 第 4 SIC の説明 */
        val fourthSicDesc: String?,
        /** 第 5 SIC コード */
        val fifthSic: String?,
        /** 第 5 SIC の説明 */
        val fifthSicDesc: String?,
        /** 第 6 SIC コード */
        val sixthSic: String?,
        /** 第 6 SIC の説明 */
        val sixthSicDesc: String?,
        /** 第 1 NAICS コード */
        val primaryNaics: String?,
        /** 第 1 NAICS の説明 */
        val primaryNaicsDesc: String?,
        /** 第 2 NAICS コード */
        val secondNaics: String?,
        /** 第 2 NAICS の説明 */
        val secondNaicsDesc: String?,
        /** 第 3 NAICS コード */
        val thirdNaics: String?,
        /** 第 3 NAICS の説明 */
        val thirdNaicsDesc: String?,
        /** 第 4 NAICS コード */
        val fourthNaics: String?,
        /** 第 4 NAICS の説明 */
        val fourthNaicsDesc: String?,
        /** 第 5 NAICS コード */
        val fifthNaics: String?,
        /** 第 5 NAICS の説明 */
        val fifthNaicsDesc: String?,
        /** 第 6 NAICS コード */
        val sixthNaics: String?,
        /** 第 6 NAICS の説明 */
        val sixthNaicsDesc: String?,
        /** 場所所有者インジケータ */
        val ownOrRent: String?,
        /** 従業員数 - 場所インジケータ */
        val employeesHereReliability: String?,
        /** 年間売上高インジケータ */
        val salesVolumeReliability: String?,
        /** 現地通貨コード */
        val currencyCode: String?,
        /** 法律上の会社形態 */
        val legalStatus: String?,
        /** 従業員数 - 合計インジケータ */
        val employeesTotalReliability: String?,
        /** マイノリティ所有インジケータ */
        val minorityOwned: String?,
        /** 女性所有インジケータ */
        val womenOwned: String?,
        /** 中小企業インジケータ */
        val smallBusiness: String?,
        /** マーケティングセグメンテーションクラスタ */
        val marketingSegmentationCluster: String?,
        /** 輸入/輸出 */
        val importExportAgent: String?,
        /** 関連子会社インジケータ */
        val subsidiary: String?,
        /** 第 2 取引形態 */
        val tradeStyle2: String?,
        /** 第 3 取引形態 */
        val tradeStyle3: String?,
        /** 第 4 取引形態 */
        val tradeStyle4: String?,
        /** 第 5 取引形態 */
        val tradeStyle5: String?,
        /** 国民 ID 番号 */
        val nationalId: String?,
        /** 国民 ID 体系 */
        val nationalIdType: String?,
        /** 米国納税 ID 番号 */
        val usTaxId: String?,
        /** 地理コードの精度 */
        val geoCodeAccuracy: String?,
        /** 支払遅延リスク */
        val marketingPreScreen: String?,
        /** グローバル代表企業の D-U-N-S Number */
        val globalUltimateDunsNumber: String?,
        /** グローバル代表企業の会社名 */
        val globalUltimateBusinessName: String?,
        /** 親会社の D-U-N-S Number */
        val parentOrHqDunsNumber: String?,
        /** 親会社の会社名 */
        val parentOrHqBusinessName: String?,
        /** 国内代表企業の D-U-N-S Number */
        val domesticUltimateDunsNumber: String?,
        /** 国内代表企業の会社名 */
        val domesticUltimateBusinessName: String?,
        /** 場所種別 */
        val locationStatus: String?,
        /** 現地通貨の ISO コード */
        val companyCurrencyIsoCode: String?,
        /** 会社の説明 */
        val description: String?,
        /** S&P 500 */
        val includedInSnP500: String?,
        /** 立地面積の精度 */
        val premisesMeasureReliability: String?,
        /** 立地面積単位 */
        val premisesMeasureUnit: String?,
        /** 第 1 SIC8 コード */
        val primarySic8: String?,
        /** 第 1 SIC8 の説明 */
        val primarySic8Desc: String?,
        /** 第 2 SIC8 コード */
        val secondSic8: String?,
        /** 第 2 SIC8 の説明  */
        val secondSic8Desc: String?,
        /** 第 3 SIC8 コード */
        val thirdSic8: String?,
        /** 第 3 SIC8 の説明 */
        val thirdSic8Desc: String?,
        /** 第 4 SIC8 コード */
        val fourthSic8: String?,
        /** 第 4 SIC8 の説明 */
        val fourthSic8Desc: String?,
        /** 第 5 SIC8 コード */
        val fifthSic8: String?,
        /** 第 5 SIC8 の説明 */
        val fifthSic8Desc: String?,
        /** 第 6 SIC8 コード */
        val sixthSic8: String?,
        /** 第 6 SIC8 の説明 */
        val sixthSic8Desc: String?
) : SObject {
    override fun title(): String = "${super.title()}"
    override fun content(): String = listOfNotNull("").joinToString("\n")
    override val objectType: SObjects = SObjects.DandBCompany
}
